core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -614,8 +612,10 @@ public long getWorkerId(WorkerNetAddress workerNetAddress) {
     }
 
     // Generate a new worker id.
-    long workerId = mNextWorkerId.getAndIncrement();
-    mWorkers.add(new MasterWorkerInfo(workerId, workerNetAddress));
+    long workerId = IdUtils.getRandomNonNegativeLong();
+    while (!mWorkers.add(new MasterWorkerInfo(workerId, workerNetAddress))) {
```

Comment:
The `IndexedSet` add will add to the primary index first, to detect if the id exists yet. If a different `workerNetAddresses` with the same `id` is added, it will conflict with the existing `id`, so the add will return `false`.